client/jmx-adapter/src/main/java/org/jolokia/client/jmxadapter/JolokiaJmxConnector.java

```diff
@@ -71,6 +67,15 @@ public void connect(Map<String, ?> env) throws IOException {
     postCreateAdapter();
   }
 
+  public String getJolokiaProtocol(Map<String, Object> env) {
```

Comment:
In the intention to make this logic public, is it clear to the caller what the argument `env` should really be?  It should be the merged env but I'm not sure if it's really a shared assumption with the users. Rather, isn't it better to make it a boolean arg `checkStub` or something like that so that the users won't need the internal details?

Also, adding javadoc for the method should definitely help.